### ISSUE TITLE
 "date" sanitizer failing on Feb 30/31

### ISSUE THREAD

**Issue by rolandtoth** [processwire/processwire-issues#185]
### Short description of the issue

"date" sanitizer doesn't return NULL on February 30/31, even if "format" option is set.

### Expected behavior

Return NULL on non-existing days.

### Actual behavior

Returns March 1/2/3, just like strtotime.

### Optional: Suggestion for a possible fix

See this forum post:

https://processwire.com/talk/topic/15508-date-sanitizer-failing-on-feb-31/?do=findComment&comment=138631

### Steps to reproduce the issue

```php
echo wire('sanitizer')->date('2017/02/27'); // 1488153600
echo wire('sanitizer')->date('2017/02/31'); // 1488499200 - FAIL
echo wire('sanitizer')->date('2017/02/32'); // NULL
```

### Setup/Environment

* ProcessWire version: 3.052
* (Optional) PHP version: 7
* (Optional) MySQL version:  5.5

**Comment by LostKobrakai** [comment]
Just to add that here: It seems the sanitizer is not even using the optional provided format and does only use strtotime().

**Comment by ryancramerdesign** [comment]
Thanks, I've pushed an update to this method. The method is really more about sanitization than validation, so if a date like 2017/02/31 can be converted to the proper date, then that's great. I wasn't aware of that strtotime behavior. But understand some may not want that behavior, so I've added a "strict" option that, when combined with a $format, makes it validate that the date matches the format and is a valid date. Also updated the method notes to reflect all this.
